### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by degisew**
### TDD project
**In this PR, we implemented the following requirements:**
- Created a class called Solver.
-  Created a method called factorial that takes one argument, an integer N, and returns the [factorial] 
    (https://en.wikipedia.org/wiki/Factorial) for that number.
-  Created a method called reverse that takes one argument, a string word, and returns word reversed (e.g. if word is "hello" it 
    returns "olleh").
- Created a method called fizzbuzz that takes one argument, an integer N, and returns a string.
-  started with tests for all of the methods described above.

### CODE REVIEW

File: solver.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class Solver
```
Comment (Ismailco) [inline solver.rb:1]: - [ ] You should create tests before implementing any methods in the Solver class, kindly make sure your project is created using TDD (tests should be created first)

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @degisew, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: solver.rb (hunk 1 above)
Comment (degisew) [reply solver.rb:1]: Okay, Thanks. we will fix it.

**Review by Munsa1 — APPROVED** [review]
Hi @degisew @Ceemos96 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**